src/powerhouse/components/drag-and-drop/draggable.tsx

```diff
@@ -0,0 +1,32 @@
+import { useDrag } from 'react-aria';
+import { CUSTOM_OBJECT_FORMAT } from './constants';
+
+export interface DraggableRenderProps {
+    isDragging: boolean;
+}
+
+export interface DraggableProps<Item = any>
+    extends Omit<React.HTMLAttributes<HTMLDivElement>, 'children'> {
+    children?: (props: DraggableRenderProps) => React.ReactNode;
+    item: Item;
+}
+
+export function Draggable<Item = any>(props: DraggableProps<Item>) {
+    const { item, children, ...divProps } = props;
+
+    if (!children) return null;
+
+    const { dragProps, isDragging } = useDrag({
+        getItems: () => [
+            {
+                [CUSTOM_OBJECT_FORMAT]: JSON.stringify(item),
+            },
+        ],
+    });
+
+    return (
+        <div {...dragProps} {...divProps}>
```

Comment:
To support keyboard navigation this has to have an ARIA role and a tabIndex="0"
In the [example](https://react-spectrum.adobe.com/react-aria/useDrag.html#example) they use `button`.

I was checking this and saw that there are aria roles for tree views: [[](url)](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tree_role).

We don't to worry about this too much right now but might be useful in the future.

Here's how the keyboard navigation works:
```React Aria supports traditional mouse and touch based drag and drop, but also implements keyboard and screen reader friendly interactions. Users can press Enter on a draggable element to enter drag and drop mode. Then, they can press Tab to navigate between drop targets, and Enter to drop or Escape to cancel. Touch screen reader users can also drag by double tapping to activate drag and drop mode, swiping between drop targets, and double tapping again to drop.```

Reply:
fixed at 47b9d51b1f51b21fe50f59ee0766e5e8967ed5dd